### PR TITLE
Add move command for machinery

### LINKED ISSUE #1397: Add a command to rename descriptions
We have `machinery copy` and we have `machinery remove`. But we don't have a command to rename a description.

Following the scheme of naming commands after the UNIX tools, this command should probably called `machinery move`.

### PR DESCRIPTION
**Pull request by keberle94**
Fixes #1397
The machinery move command moves a systemdescription to the specified
name.

### CODE REVIEW

File: lib/cli.rb [hunk 1]
```diff
@@ -404,6 +404,23 @@ def self.check_port_validity(port)
     end
   end
 
+  desc "Move system description"
+  long_desc <<-LONGDESC
+    Move a system description.
+
+    The system description is moved under the provided name.
```
Comment (thardeck) [inline lib/cli.rb]: I would prefer `The system description name is changed to the provided name.`

File: man/machinery-move.1.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  * `FROM_NAME` (required):
+    Current name of the system description.
+
+  * `TO_NAME` (required):
+    New name of the system description.
+
+
+### EXAMPLES
+
+  * Move the system description `earth` to the name `moon`:
```
Comment (thardeck) [inline man/machinery-move.1.md]: I would rephrase
```
Rename the system description `earth` to `moon`:
```

File: spec/unit/system_description_store_spec.rb [hunk 3]
```diff
@@ -176,6 +176,35 @@
...
+      create_machinery_dir
+    end
+
+    it "moves an existing SystemDescription" do
+      expect(store.list).to eq([test_name])
+      store.move(test_name, new_name)
+      expect(store.list).to eq([new_name])
+    end
+
+    it "throws an error when the to be copied SystemDescription does not exist" do
```
Comment (thardeck) [inline spec/unit/system_description_store_spec.rb]: s/copied/moved/

File: spec/unit/system_description_store_spec.rb [hunk 4]
```diff
@@ -176,6 +176,35 @@
...
+      expect {
+        store.move("foo_bar_does_not_exist", new_name)
+      }.to raise_error(Machinery::Errors::SystemDescriptionNotFound, /foo_bar_does_not_exist/)
+    end
+
+    it "throws an error when the new name already exists" do
+      store.copy(test_name, new_name)
```
Comment (thardeck) [inline spec/unit/system_description_store_spec.rb]: s/copy/move/

**Comment by keberle94** [comment]
Squashed by #1411
